src/main/webapp/dev/js/common/instructor.js

```diff
@@ -358,14 +358,18 @@ function bindCourseDeleteAllLinks() {
         event.preventDefault();
 
         const $clickedLink = $(event.currentTarget);
-        const messageText = 'Are you sure you want to permanently delete all the courses? '
-                + 'This operation will delete all students and sessions in these courses. '
-                + 'All instructors of these courses will not be able to access them hereafter as well.';
+        const softDeletedCourses = $('[data-soft-deleted-course-id]').toArray();
+        const messageText = `<p>Are you sure you want to permanently delete the following ${softDeletedCourses.length} `
+                + 'courses in the <b>Recycle Bin</b>? This operation will delete all students and sessions in these '
+                + 'courses. All instructors of these courses will not be able to access them hereafter as well.</p>'
+                + '<ul>'
+                + `${softDeletedCourses.map(softDeletedCourse => `<li>${$(softDeletedCourse).text()}`).join('</li>')}`
```

Comment:
We can have a higher-level abstraction here such as `transformArrayToList` to improve readability.

Also, we want to map each array element to list elements, joining a half-built list element with a closing `li` tag is a weird abstraction and not very understandable. And what happens to the last list element?

Also, not your fault but we have a missing `map` polyfill. I'm leaving a note here for now so we can create an issue to add that via `core-js` in `onStart.js` later.

Reply:
Any suggestion on the format?
```javascript
function transformArrayToUnOrderedListHtml(softDeletedCourses) {
    return `<ul>${softDeletedCourses
            .map(softDeletedCourse => `<li>${$(softDeletedCourse).data('softDeletedCourseId')}</li>`)
            .join('')}</ul>`;
}
```

Reply:
How about something like this:
```Javascript
/**
 * Returns the HTML required to render the soft deleted courses as an unordered list.
 */
function generateUnorderedListHtml(softDeletedCourses) {
    return `\
<ul>
${softDeletedCourses
            .map(softDeletedCourse => `<li>${$(softDeletedCourse).data('softDeletedCourseId')}</li>`)
            .join('')}
</ul>`;
}
```

Reply:
Will do.